spec/install/gemfile/path_spec.rb

```diff
@@ -13,6 +13,21 @@
     should_be_installed("foo 1.0")
   end
 
+  it "resolves from prior lockfile for source path gems" do
+    build_lib "rack-obama", "1.0", :path => lib_path("same_path")
+    build_lib "net-ssh", "1.0",    :path => lib_path("same_path")
+
+    install_gemfile <<-G
+      path "#{lib_path("same_path")}/" do
+        gem 'rack-obama'
+        gem 'net-ssh'
```

Comment:
@segiddins I tried a few different combinations of existing gems for the test suite but not every pair of path based gems cause it to fail.  I don't know why the array order are not consistent.  

Keep me honest here, I think I did this test correctly.